# 05 EDA — Payments

Read-only inspection of `olist_order_payments_dataset.csv`. No cleaning, charts, or feature engineering.

## 1. Import libraries

Import pandas for tabular inspection.

In [1]:
import pandas as pd

**Observation:** pandas is ready.

## 2. Load CSV

Load only this CSV into memory without changing the source.

In [2]:
file_path = '../data/raw/olist/olist_order_payments_dataset.csv'
df = pd.read_csv(file_path)

**Observation:** The table is loaded; the raw CSV is unchanged.

## 3. Shape

Check table size.

In [3]:
df.shape

(103886, 5)

**Observation:** This reports the table's row and column count.

## 4. First 5 rows

Preview the raw structure.

In [4]:
df.head()

                           order_id  ...  payment_value
0  b81ef226f3fe1789b1e8b2acac839d17  ...          99.33
1  a9810da82917af2d9aefd1278f1dcfa0  ...          24.39
2  25e8ea4e93396b6fa0d3dd708e76c1bd  ...          65.71
3  ba78997921bbcdc1373bb41e913ab953  ...         107.78
4  42fdf880ba16b47b59251dd489d4441a  ...         128.45

[5 rows x 5 columns]

**Observation:** The preview confirms the row grain and field layout.

## 5. Column names

List available fields.

In [5]:
df.columns.tolist()

['order_id', 'payment_sequential', 'payment_type', 'payment_installments', 'payment_value']

**Observation:** These are the fields available for this table's analysis.

## 6. Data types

Review pandas data types.

In [6]:
df.dtypes

order_id                    str
payment_sequential        int64
payment_type                str
payment_installments      int64
payment_value           float64
dtype: object

**Observation:** Types show which fields are text, numeric, or converted datetime values.

## 7. Missing values

Count nulls by field.

In [7]:
df.isna().sum()

order_id                0
payment_sequential      0
payment_type            0
payment_installments    0
payment_value           0
dtype: int64

**Observation:** Missing counts identify incomplete fields without changing them.

## 8. Missing percentage

Measure missingness relative to table size.

In [8]:
(df.isna().mean().mul(100).round(2)).sort_values(ascending=False)

order_id                0.0
payment_sequential      0.0
payment_type            0.0
payment_installments    0.0
payment_value           0.0
dtype: float64

**Observation:** Percentages show the relative scale of incomplete fields.

## 9. Duplicate rows

Count complete row duplicates.

In [9]:
df.duplicated().sum()

np.int64(0)

**Observation:** This is the number of repeated full rows; none are removed.

## 10. Duplicate primary keys

Validate the documented candidate key.

In [10]:
primary_key = ['order_id', 'payment_sequential']
df.duplicated(subset=primary_key, keep=False).sum()

np.int64(0)

**Observation:** This counts rows involved in duplicate candidate keys for order_id, payment_sequential.

## 11. Unique values

Inspect meaningful category or entity counts.

In [11]:
df['payment_type'].value_counts(dropna=False)

payment_type
credit_card    76795
boleto         19784
voucher         5775
debit_card      1529
not_defined        3
Name: count, dtype: int64

**Observation:** This gives a concise view of important distinct values.

## 12. Basic describe()

Generate a basic summary for all fields.

In [12]:
df.describe(include='all')

                                order_id  ...  payment_value
count                             103886  ...  103886.000000
unique                             99440  ...            NaN
top     fa65dad1b0e818e3ccc5cb0e39231352  ...            NaN
freq                                  29  ...            NaN
mean                                 NaN  ...     154.100380
std                                  NaN  ...     217.494064
min                                  NaN  ...       0.000000
25%                                  NaN  ...      56.790000
50%                                  NaN  ...     100.000000
75%                                  NaN  ...     171.837500
max                                  NaN  ...   13664.080000

[11 rows x 5 columns]

**Observation:** The summary describes completeness, distinctness, common values, and numeric ranges where applicable.

## 13. Table-specific observations

The table records 103,886 payment events for 99,440 orders. Multiple rows per order represent sequential or split payments.

## 14. Data quality observations

No missing or exact duplicate rows were found. The composite payment key is valid and every payment references an existing order; one order has no payment row.